scipy/stats/_continuous_distns.py

```diff
@@ -8439,76 +8438,92 @@ def _rvs(self, chi, size=None, random_state=None):
-        # samples of Y and convert back to X
+        # In both cases, the cdf G of g can be easily computed to apply the
+        # rejection method. We use case 1 for chi < 1,
+        # case 2 for 1 <= chi <= 1.825
         #
-        # The expected number of iterations using the rejection method
-        # increases with increasing chi, whereas the expected number of
-        # iterations using the ratio of uniforms method decreases with
-        # increasing chi. The crossover occurs where
-        # chi*(1 - exp(-0.5*chi**2)) = 8*sqrt(2)*exp(-1.5) => chi ~ 2.611
-        # Switching algorithms at chi=2.611 means that the expected number of
-        # iterations is always below 2.2.
-
-        if chi <= 2.611:
-            # use rejection method
-            size1d = tuple(np.atleast_1d(numsamples))
-            N = int(np.prod(size1d))
+        # if chi > 1.825:
+        # use relation to the Gamma distribution: if X is ARGUS with parameter
+        # chi), then Y = chi**2 * (1 - X**2) / 2 has density proportional to
+        # sqrt(u) * exp(-u) on [0, chi**2 / 2], i.e. a Gamma(3/2) distribution
+        # conditioned on [0, chi**2 / 2]).
+        #
+        # The points 1 and 1.825 are determined by a comparison of the
+        # runtime of the different methods
+
+        size1d = tuple(np.atleast_1d(numsamples))
+        N = int(np.prod(size1d))
+
+        if isinstance(random_state, np.random.RandomState) and chi <= 1.825:
             x = np.zeros(N)
-            echi = np.exp(-chi**2 / 2)
             simulated = 0
+            chi2 = chi * chi
+            if chi < 1:
+                d = -chi2 / 2
+                while simulated < N:
+                    k = N - simulated
+                    u = random_state.uniform(size=k)
+                    v = random_state.uniform(size=k)
+                    z = v**(2/3)
+                    accept = (np.log(u) <= d * z)
+                    num_accept = np.sum(accept)
+                    if num_accept > 0:
+                        rvs = np.sqrt(1 - z[accept])
+                        x[simulated:(simulated + num_accept)] = rvs
+                        simulated += num_accept
+            elif chi <= 1.825:
+                echi = np.exp(-chi2 / 2)
+                while simulated < N:
+                    k = N - simulated
+                    u = random_state.uniform(size=k)
+                    v = random_state.uniform(size=k)
+                    z = 2 * np.log(echi * (1 - v) + v) / chi2
+                    accept = (u**2 + z <= 0)
+                    num_accept = np.sum(accept)
+                    if num_accept > 0:
+                        rvs = np.sqrt(1 + z[accept])
+                        x[simulated:(simulated + num_accept)] = rvs
+                        simulated += num_accept
+        elif isinstance(random_state, np.random.Generator) and chi < 5:
+            if chi < 1.15:
+                x = _stats._rvs_argus_rejection_beta(chi, N, random_state)
```

Comment:
The default random_state is still a `RandomState`, not a `Generator`, and I'm guessing that won't change in the near future. Thus, the calls to this `_rvs` function *mostly* won't use this cython path. As such, is it worth adding the cython code?